v2/lints/cabf_br/lint_ext_tor_service_descriptor_hash_invalid.go

```diff
@@ -184,14 +186,19 @@ func (l *torServiceDescHashInvalid) Execute(c *x509.Certificate) *lint.LintResul
 		descriptorMap[hostname] = descriptor
 	}
 
-	// Check if any of the onion subjects in the certificate don't have
-	// a TorServiceDescriptorHash for the eTLD+1 in the descriptorMap.
-	for eTLDPlusOne, subjDomain := range onionETLDPlusOneMap {
-		if _, found := descriptorMap[eTLDPlusOne]; !found {
-			return failResult(
-				"%s subject domain name %q does not have a corresponding "+
-					"TorServiceDescriptorHash for its eTLD+1",
-				onionTLD, subjDomain)
+	// For EV certificates, every `.onion` name is required to have a
+	// TorServiceDescriptorHash, so check if any of the onion subjects in the
+	// certificate don't have a TorServiceDescriptorHash for the eTLD+1 in the
+	// descriptorMap.
+	// See also https://github.com/cabforum/documents/issues/190
```

Comment:
We don't really have any normative guidance on what to do if this extension appears in a non-EV cert. My interpretation is that despite it being unnecessary for V3 certs, because it's optional, it should be legal to include this for some but not all v3 onion names.

Similarly, if/when we fix it for EV, I imagine that it'll be required for V2 names, but optional for V3 names. While prohibited-for-v3 makes the linting slightly easier, I suspect it makes CAs' lives harder, so I doubt that'll happen in the EVGs.